DALOptimizer/MatchInvocation.cs

```diff
@@ -0,0 +1,157 @@
+using ICSharpCode.NRefactory.Editor;
+using ICSharpCode.NRefactory.PatternMatching;
+using ICSharpCode.NRefactory.Semantics;
+using ICSharpCode.NRefactory.TypeSystem;
+
+namespace DALOptimizer
+{
+    class MatchInvocation
+    {
+        public const string EXPRESSIONSTATEMENT = "ExpressionStatement";
+        public const string CATCHCLAUSE = "CatchClause";
+        public const string FIELDDECLARATION = "FieldDeclaration";
+        public const string VARIABLEDECLSTMT = "VariableDeclarationStatement";
+        public const string ASSIGNMENTEXPRESSION = "AssignmentExpression";
+        public const string PROPERTYDECLARATION = "PropertyDeclaration";
+        public const string BLOCKSTATEMENT = "BlockStatement";
+        public const string METHODDECLARARION = "MethodDeclaration";
+        public const string TYPEDECLARATION = "TypeDeclaration";
+
+        AllPatterns Pat = new AllPatterns();
+
+        public void CheckAllInvocation(CSharpFile file, CSharpAstResolver astResolver) 
+        {
+            foreach (var invocation in file.syntaxTree.Descendants.OfType<AstNode>())
+            {
+                //track all expression Statemens
+                if (invocation.GetType().Name == EXPRESSIONSTATEMENT)
+                {
+                    MatchExprStmt(invocation, file, astResolver);
+                    continue;
+                }
+                //catch clause
+                if (invocation.GetType().Name == CATCHCLAUSE)
+                {
+                    MatchCatchClause(invocation, file, astResolver);
+                    continue;
+                }
+
+                // For All Global Field Declarations
+                if (invocation.GetType().Name == FIELDDECLARATION)
+                {
+                    MatchFieldDecl(invocation, file, astResolver);
+                    continue;
+                }
+                // For variable Decaration of type {SqlCommand cmd = new SqlCommand ("dbo.InboxDeviceReport", con);}
+                if (invocation.GetType().Name == VARIABLEDECLSTMT)
+                {
+                    MatchVarDeclStmt(invocation, file, astResolver);
+                    continue;
+                }
+
+                if (invocation.GetType().Name == ASSIGNMENTEXPRESSION)
+                {
+                    MatchAssExpr(invocation, file, astResolver);
+                    continue;
+                }
+
+                //check Global Property Declaration
+                if (invocation.GetType().Name == PROPERTYDECLARATION)
+                {
+                    MatchPropDecl(invocation, file, astResolver);
+                    continue;
+                }
+
+                //check Finally Block
+                if (invocation.GetType().Name == BLOCKSTATEMENT)
+                {
+                    MatchBlock(invocation, file, astResolver);
+                    continue;
+                }
+
+                if (invocation.GetType().Name == METHODDECLARARION)
```

Comment:
@sushantyelpale no need to calculate `invocation.GetType().Nam` for all the `if` condition checks.
Calculate it once and then use `switch` case to perform the necessary operation.